### PR TITLE
Bump schnorrkel to 0.8.3

### PR DESCRIPTION
**Pull request by jacogr**
@bkchr This is intended to merge into your upgrade branch.

- allows for verification of 0.1.1 and 0.8.x signatures
- aligns derivation with 0.1.1
- aligns expansion of pairs with 0.1.1

### CODE REVIEW

File: subkey/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ clap = { version = "~2.32", features = ["yaml"] }
 tiny-bip39 = "0.6.0"
 rustc-hex = "2.0"
 substrate-bip39 = "0.3.1"
-schnorrkel = "0.8.0"
+schnorrkel = { version = "0.8.3", features = ["preaudit_deprecated"] }
```
Comment (bkchr) [inline subkey/Cargo.toml]:
```suggestion
```

It is not used :D

File: core/primitives/src/sr25519.rs [hunk 2]
```diff
@@ -476,29 +476,22 @@ impl TraitPair for Pair {
 
 	/// Verify a signature on a message. Returns true if the signature is good.
 	fn verify<P: AsRef<Self::Public>, M: AsRef<[u8]>>(sig: &Self::Signature, message: M, pubkey: P) -> bool {
-		let signature: schnorrkel::Signature = match schnorrkel::Signature::from_bytes(&sig.as_ref()) {
-			Ok(some_signature) => some_signature,
-			Err(_) => return false
-		};
+		// Match both schnorrkel 0.1.1 and 0.8.0+ signatures, supporting both wallets
```
Comment (bkchr) [inline core/primitives/src/sr25519.rs]: I think we should open an issue for this? So we don't forget to switch.

Comment (burdges) [reply core/primitives/src/sr25519.rs]: Probably.  And/or on schnorrkel.  I'll remove the preaudit_deprecated feature entirely whenever all testnets migrate.

Comment (bkchr) [reply core/primitives/src/sr25519.rs]: Okay, yeah then let's add this issue to schnorrkel.

Comment (burdges) [reply core/primitives/src/sr25519.rs]: https://github.com/w3f/schnorrkel/issues/42

**Pull request merged**